MOKApp/app/src/main/java/mok/it/app/mokapp/fragments/MyBadgesFragment.kt

```diff
@@ -33,40 +34,33 @@ class MyBadgesFragment :
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        getProjects(userModel.collectedBadges)
-    }
-
-    private fun getProjects(projectIds: List<String>?) {
-        completedProjects = ArrayList()
-        initRecyclerView()
-        projectIds?.forEach {
-            val docRef = Firebase.firestore.collection(Collections.projects).document(it)
-            docRef.get()
-                .addOnSuccessListener { document ->
-                    if (document != null) {
-                        val badge = document.toObject(Project::class.java)!!
-                        completedProjects.add(badge)
-                        if (projectIds.size == completedProjects.size) {
-                            initRecyclerView()
-                        }
-                    }
+        projectService.getProjectsByIds(
+            projectIds = userModel.projectBadges.keys.toList(),
+            onComplete = { projectsList ->
+                collectedBadges.clear()
+                for (project in projectsList) {
+                    collectedBadges.add(Pair(project, userModel.projectBadges[project.id] ?: 0))
```

Comment:
A collected badges új megfelelője nem egy completed projects kellene hogy legyen? Vagy szándékos, hogy a mancsaimban az is megjelenik, amin az illető még nem szerzett mancsot?

Reply:
Szándékos, ez az a felület lenne, ahol minden projekted látszódik, amire fel vagy jelentkezve. És adott esetben hoogy hány mancsnál tartasz éppen rajta.